### DISCUSSION TITLE
Can I set a particle boundary for embedded boundary?

### DISCUSSION THREAD

**Discussion by Kellysherry** [BLAST-WarpX/warpx#4703] [Q&A]
Hi,
I wanted to make a simple simulation on electron extraction, the electrons will be injected from the bottom boundary whose potential is -1000 V and there is a grid in the calculation domain which is set to 100 V.
But it failed to converge and I found that if I dismiss the grid this problem will be solved. 
I guess that's because I didn't set a particle boundary for the grid but I didn't find any input parameters about it.

**Comment by roelof-groenewald** [comment]
Hi @Kellysherry. Currently any embedded boundaries in WarpX will be absorbing for particles. There are ways to have more involved interactions between the particles and the embedded boundary but currently these require handling such interactions via python callbacks. See for example this simulation: https://github.com/ECP-WarpX/WarpX/tree/development/Examples/Physics_applications/spacecraft_charging.

**Reply by Kellysherry** [reply]
Thanks for your reply, the absorbing condition will be fine for me. But I found that as the embedded boundary interacts with the surrounding particles, its electric potential changes, unless setting algo.maxwell_solver=none. Is it because of the absorbing condition or I missed some parameters somewhere else?
